master/internal/api_trials.go

```diff
@@ -105,11 +104,19 @@ type trialAllocation struct {
 	Task     model.TaskID
 }
 
+func getLatestTaskIDFromTrialProto(t *trialv1.Trial) model.TaskID {
+	if len(t.TaskIds) == 0 {
+		panic(fmt.Sprintf("trial proto object %+v was without associated task", t))
```

Comment:
Right now we will always have a task ID since we had a not null constraint on trials and whenever we insert a trial we require a task ID. 

This isn't enforced by the database technically, is there a way to add a constraint so that trial rows can't be inserted without also inserting into trial_id_task_id?